models/world_coordinates.py

```diff
@@ -152,4 +120,72 @@ def get_downstreams(self, session=None):
 
         downstreams = zps + subs
         return downstreams
+
+    def save( self, filename=None, **kwargs ):
+        """Write the WCS data to disk.
+        Updates self.filepath
+        Parameters
+        ----------
+          filename: str or path
+             The path to the file to write, relative to the local store
+             root.  Do not include the extension (e.g. '.psf') at the
+             end of the name; that will be added automatically.
+             If None, will call image.invent_filepath() to get a
+             filestore-standard filename and directory.
+          Additional arguments are passed on to FileOnDiskMixin.save
+        """ 
+
+        # ----- Make sure we have a path ----- #
+        # if filename already exists, check it is correct and use
+
+        if filename is not None:
+            if not filename.endswith('.txt'):
+                filename += '.txt'
+            self.filepath = filename
+
+        # if not, generate one
+        else:
+            if self.image.filepath is not None:
+                self.filepath = self.image.filepath
+            else:
+                self.filepath = self.image.invent_filepath()
+
+            if self.provenance is None:
+                raise RuntimeError("Can't invent a filepath for the WCS without a provenance")
+            self.filepath += f'.wcs_{self.provenance.id[:6]}.txt'
+
+        txtpath = pathlib.Path( self.local_path ) / f'{self.filepath}'
+
+        # ----- Get the header string to save and save ----- #
+        header_txt = self.wcs.to_header().tostring(padding=False, sep='\\n' )
+
+        if txtpath.exists():
+            if ( 'overwrite' in kwargs and not kwargs['overwrite'] ):
```

Comment:
```suggestion
            if kwargs.get('overwrite', True):
```

Reply:
I wrote this check to raise the error if overwrite exists and is explicitly set to 'False', so I think this suggestion reverses the logic, though I like the syntax used in yours.
I think I'll use
`if kwargs.get('overwrite', True) == False:`
which I like a little better for reading clarity than
`if not kwargs.get('overwrite', True):`